client/tests/phpunit/Security/DocumentVoterTest.php

```diff
@@ -0,0 +1,232 @@
+<?php declare(strict_types=1);
+
+namespace AppBundle\Security;
+
+use AppBundle\Entity\Client;
+use AppBundle\Entity\Organisation;
+use AppBundle\Entity\Report\Document;
+use AppBundle\Entity\Report\Report;
+use AppBundle\Entity\User;
+use Doctrine\Common\Collections\ArrayCollection;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+
+class DocumentVoterTest extends TestCase
+{
+    private $sut;
+    private $token;
+    private $deputy;
+    private $organisation;
+    private $client;
+    private $report;
+    private $document;
+
+    public function setUp(): void
+    {
+        $this->sut = new DocumentVoter();
+        $this->token = $this->createMock(TokenInterface::class);
+
+        $this->deputy = (new User())->setId(87);
+        $this->organisation =  (new Organisation())->setId(31)->setIsActivated(true);
+        $this->report = new Report();
+        $this->client = new Client();
+        $this->document = new Document();
+    }
+
+    /**
+     * @dataProvider getSupportedAttributes
+     * @param string $attribute
+     * @param bool $expected
+     */
+    public function testSupports(string $attribute, int $expected): void
+    {
+        $this->token->method('getUser')->willReturn(null);
+        $this->assertEquals($expected, $this->sut->vote($this->token, new Report(), [$attribute]));
+    }
+
+    /**
+     * @return array
+     */
+    public function getSupportedAttributes(): array
+    {
+        return [
+            [DocumentVoter::ADD_DOCUMENT, Voter::ACCESS_DENIED],
+            [DocumentVoter::DELETE_DOCUMENT, Voter::ACCESS_DENIED],
+            ['UNKNOWN', Voter::ACCESS_ABSTAIN],
+        ];
+    }
+
+    /**
+     * @test
+     */
+    public function voteOnAttributeAllowsLayDeputiesToAddDocumentsToTheirOwnReport(): void
+    {
+        $this->token->method('getUser')->willReturn($this->deputy);
+
+        $this
+            ->ensureReportBelongsToClient()
+            ->ensureClientBelongsToDeputy()
```

Comment:
Need to remember this style of setting up test data - I find it really easy to read compared to generator functions